cmd/main.go

```diff
@@ -73,6 +75,24 @@ func main() {
 	pflag.CommandLine.StringVar(&mgrOptions.PprofBindAddress, "profiler-address", "",
 		"Bind address to expose the pprof profiler (e.g. localhost:6060)")
 
+	pflag.CommandLine.BoolVar(&mgrOptions.LeaderElection, "leader-elect", mgrOptions.LeaderElection,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
+
+	pflag.CommandLine.StringVar(
+		&mgrOptions.LeaderElectionID,
+		"leader-election-id",
+		mgrOptions.LeaderElectionID,
+		"The name of the resource that leader election will use for holding the leader lock.",
+	)
```

Comment:
We seem to be setting the default to the value that's being set. While in the struct declaration, we have not set a value at all. Shouldn't we set a sensible default here? Also, it'd definitely be cleaner if the default value is actually pointing to a const declared elsewhere instead.

Reply:
Setting a default here can lead to conflicts if we run multiple deployments in the same cluster/namespace. Although this is unlikely, it's possible. I will default it to the empty string here instead. It is a required flag if leader election is enabled.

Reply:
And it will already use a default for the flag as part of the generated CAPI IPAM manifests when deploying through `clusterctl`.